src/Core/Managed/Shared/Extensibility/Implementation/External/Tags.cs

```diff
@@ -147,38 +64,29 @@ internal static void SetDateTimeOffsetValueOrRemove(this IDictionary<string, str
             }
         }
 
-        internal static void SetGuidValueOrRemove(this IDictionary<string, string> tags, string tagKey, Guid? tagValue)
-        {
-            if (tagValue == null)
-            {
-                SetTagValueOrRemove(tags, tagKey, tagValue);
-            }
-            else
-            {
-                string tagValueString = tagValue.Value.ToString();
-                SetTagValueOrRemove(tags, tagKey, tagValueString);
-            }
-        }
-
         internal static void SetTagValueOrRemove<T>(this IDictionary<string, string> tags, string tagKey, T tagValue)
         {
             SetTagValueOrRemove(tags, tagKey, Convert.ToString(tagValue, CultureInfo.InvariantCulture));
         }
 
-        internal static void InitializeTagValue<T>(this IDictionary<string, string> tags, string tagKey, T tagValue)
+        internal static string CopyTagValue(string sourceValue, string targetValue)
```

Comment:
Consider this form:
```C#
internal static void CopyTagValue(string srouceValue, ref string target)
{
    if (!string.IsNullOrEmpty(sourceValue) || string.IsNullOrEmpty(target))
    {
        target = sourceValue;
    }
}
```

Then callers do this:
```Tags.CopyTagValue(other.FooTag, ref this.FooTag)```

Since the assignment is done within the body of CopyTagValue, that reduces the code that callers have to write (and eliminates one potential source of bugs where they assign to the wrong field) and it eliminates the write if the value is not to be updated (small perf win).